src/main/java/core/basesyntax/model/FruitTransaction.java

```
@@ -0,0 +1,34 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
```

Constructor assigns values without validation. You must validate inputs and throw informative RuntimeException/FruitShopException when invalid (operation == null, fruit null/blank, quantity < 0). Quote from checklist: "Remember that your code will need to be tested, so try to anticipate and handle all invalid input data in advance." Add checks at the start of the constructor and fail fast. See lines 8–11.